lib/reek/ast/object_refs.rb

```diff
@@ -1,33 +1,30 @@
 module Reek
+  # @api private
   module AST
+    ObjectRef = Struct.new(:name)
     #
     # Manages and counts the references out of a method to other objects.
     #
-    # @api private
-    class ObjectRefs  # :nodoc:
+    class ObjectRefs
       def initialize
-        @refs = Hash.new(0)
+        @refs = Hash.new { |refs, name| refs[name] = [] }
       end
 
-      def record_reference_to(exp)
-        @refs[exp] += 1
+      def most_popular
+        max = @refs.values.map(&:size).max
+        @refs.select { |_name, refs| refs.size == max }
       end
 
-      def references_to(exp)
-        @refs[exp]
+      def record_reference_to(name)
+        @refs[name] << ObjectRef.new(name)
```

Comment:
Wait a sec.
ObjectRef just takes the same name. So if we reference, say, "foo" three times "@refs" would be (kind of pseudocode):

```Ruby
"foo" => [ ObjectRef("foo"), ObjectRef("foo"), ObjectRef("foo")]
````

So isn't this just a counter? If yes, couldn't we just drop this whole ObjectRef thingie and replace that via a plain integer count:


```Ruby
@refs[name] += 1
````

and change the initialization to:

```Ruby
@refs = Hash.new { |refs, name| refs[name] = 0 }
```

?